src/converter/internal/compat/backendapi.cpp

```diff
@@ -738,6 +748,34 @@ Ret BackendApi::applyTranspose(const INotationPtr notation, const std::string& o
     return ok ? make_ret(Ret::Code::Ok) : make_ret(Ret::Code::InternalError);
 }
 
+void BackendApi::switchToPageView(IMasterNotationPtr masterNotation)
+{
+    //! NOTE: All operations must be done in page view mode
+    masterNotation->notation()->setViewMode(ViewMode::PAGE);
+    for (IExcerptNotationPtr excerpt : masterNotation->excerpts().val) {
+        excerpt->notation()->setViewMode(ViewMode::PAGE);
+    }
+}
+
+void BackendApi::renderExcerptsContents(IMasterNotationPtr masterNotation)
+{
+    //! NOTE: Due to optimization, only the master score is layouted
+    //!       Let's layout all the scores of the excerpts
+    for (IExcerptNotationPtr excerpt : masterNotation->excerpts().val) {
+        Score* score = excerpt->notation()->elements()->msScore();
+        score->setLayoutAll();
+        score->update();
+    }
+}
+
+void BackendApi::initPotentialExcerpts(notation::IMasterNotationPtr masterNotation)
+{
+    ExcerptNotationList potentialExcerpts = masterNotation->potentialExcerpts();
+
+    masterNotation->addExcerpts(potentialExcerpts);
```

Comment:
You can use MasterNotation::initExcerpt instead of adding these excerpts to the main score